### ISSUE TITLE
poetry install downgrades numpy and fails

### ISSUE THREAD

**Issue by pllim** [Cosmoglobe/zodipy#10]
**Describe the bug**

Trying to install using `poetry install` as documented but it fails with:

```
- Downgrading numpy (1.26.4 -> 1.24.4): Failed
```

**Expected behavior**

Just use installed numpy if compatible. Any reason why this is incompatible with numpy 1.26? There is no mention of such incompatibility in README.

**Desktop (please complete the following information):**
 - OS: Ubuntu (WSL2 on Windows 11)
 - Browser [e.g. chrome, safari] N/A
 - Version [e.g. 22] N/A

**Additional context**

This is part of:

* https://github.com/pyOpenSci/software-submission/issues/161

**Comment by MetinSa** [comment]
If you have time, could you check if this works as intended on the `drop_py38` branch? If so, is it sufficient to just release a >=3.9 version as a fix to this?

**Comment by pllim** [comment]
Seems to work on `drop_py38` but only after I run `poetry lock` also. Otherwise, I get this:

```
$ poetry install
Installing dependencies from lock file

pyproject.toml changed significantly since poetry.lock was last generated. Run `poetry lock [--no-update]` to fix the lock file.
```

Hope this helps!

**Comment by MetinSa** [comment]
@pllim hopefully this is resolved in the latest merge where python 3.8 is dropped and the `pyproject.toml`and the lockfile are updated, but feel free reopen this issue if you notice any issues!
